### PR TITLE
fix(marshal): Check if input is an error before checking if it is remotable

### LINKED ISSUE #1318: marshal: Errors with no own properties are mistreated as remotables
```js
const err = Error();
for (const k of ownKeys(hairlessError)) {
  delete err[k];
}
passStyleOf(harden(err)); // => Error: "[Symbol(passStyle)]" property expected: {}
```

### PR DESCRIPTION
**Pull request by gibson042**
Split from #1313 for expediency.

Fixes #1318

### CODE REVIEW

File: packages/marshal/src/passStyleOf.js [hunk 1]
```diff
@@ -194,8 +194,8 @@ export const passStyleOf = makePassStyleOf([
   CopyArrayHelper,
   CopyRecordHelper,
   TaggedHelper,
-  RemotableHelper,
   ErrorHelper,
+  RemotableHelper,
```
Comment (erights) [inline packages/marshal/src/passStyleOf.js:198]: Please also swap lines 11,12. Likewise lines 41,42. Just to maintain esthetic parallelism.

**Review by erights — APPROVED** [review]
LGTM

File: packages/marshal/src/passStyleOf.js (hunk 1 above)
Comment (gibson042) [reply packages/marshal/src/passStyleOf.js:198]: Done.

File: packages/marshal/test/test-passStyleOf.js [hunk 2]
```diff
@@ -25,6 +25,12 @@ test('passStyleOf basic success cases', t => {
   t.is(passStyleOf(harden({ then: 'non-function then ok' })), 'copyRecord');
   t.is(passStyleOf(makeTagged('unknown', undefined)), 'tagged');
   t.is(passStyleOf(harden(Error('ok'))), 'error');
+
+  const hairlessError = Error('hairless');
+  for (const k of ownKeys(hairlessError)) {
+    delete hairlessError[k];
+  }
```
Comment (kriskowal) [inline packages/marshal/test/test-passStyleOf.js:32]: # 💇

File: packages/marshal/test/test-passStyleOf.js [hunk 3]
```diff
@@ -25,6 +25,12 @@ test('passStyleOf basic success cases', t => {
   t.is(passStyleOf(harden({ then: 'non-function then ok' })), 'copyRecord');
   t.is(passStyleOf(makeTagged('unknown', undefined)), 'tagged');
   t.is(passStyleOf(harden(Error('ok'))), 'error');
+
+  const hairlessError = Error('hairless');
```
Comment (kriskowal) [inline packages/marshal/test/test-passStyleOf.js:29]: # 🙋

File: packages/marshal/test/test-passStyleOf.js [hunk 4]
```diff
@@ -25,6 +25,12 @@ test('passStyleOf basic success cases', t => {
...
+  for (const k of ownKeys(hairlessError)) {
+    delete hairlessError[k];
+  }
+  t.is(passStyleOf(harden(hairlessError)), 'error');
```
Comment (kriskowal) [inline packages/marshal/test/test-passStyleOf.js:33]: # 🧑‍🦲

**Review by kriskowal — APPROVED** [review]

**Pull request merged**